internal/cleanup/cleanup_job.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package cleanup
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/scheduler"
+)
+
+type cleanupJob struct {
+	r db.Reader
+	w db.Writer
+}
+
+func newCleanupJob(ctx context.Context, r db.Reader, w db.Writer) (*cleanupJob, error) {
+	const op = "cleanupJob.newCleanupJob"
+	switch {
+	case r == nil:
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "missing db.Reader")
+	case w == nil:
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "missing db.Writer")
+	}
+
+	return &cleanupJob{
+		r: r,
+		w: w,
+	}, nil
+}
+
+// Status reports the job’s current status.
+func (c *cleanupJob) Status() scheduler.JobStatus {
+	return scheduler.JobStatus{}
+}
+
+// Run performs the required work depending on the implementation.
+// The context is used to notify the job that it should exit early.
+func (c *cleanupJob) Run(ctx context.Context) error {
+	const op = "cleanup.(cleanupJob).Run"
+	if _, err := c.w.DoTx(ctx, db.StdRetryCnt, db.ExpBackoff{},
```

Comment:
Wrapping the exec in a transaction gives us automatic retries, but since this job just reruns every now and then on its own, maybe we can just skip the transaction?